src/main/java/org/apache/pulsar/io/sftp/sink/MessageToParquetFileOBSWriter.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.sftp.sink;
+
+import com.obs.services.ObsClient;
+import com.obs.services.ObsConfiguration;
+import com.obs.services.model.PutObjectRequest;
+import com.obs.services.model.PutObjectResult;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.parquet.column.ParquetProperties;
+import org.apache.parquet.example.data.Group;
+import org.apache.parquet.example.data.simple.SimpleGroupFactory;
+import org.apache.parquet.hadoop.ParquetFileWriter;
+import org.apache.parquet.hadoop.ParquetWriter;
+import org.apache.parquet.hadoop.example.ExampleParquetWriter;
+import org.apache.parquet.hadoop.example.GroupWriteSupport;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.MessageTypeParser;
+import org.apache.pulsar.functions.api.Record;
+import org.apache.pulsar.io.sftp.utils.Constants;
+import org.apache.pulsar.io.sftp.utils.HWObsUtil;
+
+/**
+ * Write files to local disk in parquet format.
+ */
+@Slf4j
+public class MessageToParquetFileOBSWriter implements MessageOBSWriter<byte[]> {
+
+    @Override
+    public void writeToStorage(Record<byte[]> record, OBSSink obsSink, ObsConfiguration conf) {
+        OBSSinkConfig sinkConfig = obsSink.getOBSSinkConfig();
+        String outDirectory = sinkConfig.getOutDirectory();
+        //TODO
+        String parquetFileName = "file-" + new Date().getTime() + ".parquet";
+        // The file is temporarily stored in the server's directory and will be deleted after uploading obs
+        String tempParquetFilePath = "/tmp" + outDirectory + "/" + parquetFileName;
```

Comment:
May need to be careful if inside a pod

Reply:
Can the temporary path of this file be specified by the user?